### PR TITLE
Direct parameter access: Remove some code duplication

### PR DESCRIPTION
**Pull request by dawagner**
Use templates and method overloading to simplify the code and remove a lot of
duplication in CParameterHandle.

Signed-off-by: David Wagner <david.wagner@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-146866607%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-147640900%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-147968035%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-148705875%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-148706080%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/276%23discussion_r41889114%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/276%23issuecomment-146866607%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20Please%20review%22%2C%20%22created_at%22%3A%20%222015-10-09T13%3A06%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40tcahuzax%20please%20review%20too.%22%2C%20%22created_at%22%3A%20%222015-10-13T08%3A11%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-14T08%3A05%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20Please%20review.%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A39%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A40%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%206088ac955775d6c0c2c240df5556d5c871171c7f%20parameter/ParameterHandle.cpp%20111%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/276%23discussion_r41889114%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20this%20%60true%60%20the%20only%20difference%20between%20%60getAsArrayOf%60%20and%20%60getAs%60%20%3F%20If%20so%2C%20consider%20merging%20the%20two.%22%2C%20%22created_at%22%3A%20%222015-10-13T16%3A18%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterHandle.cpp%3AL74-264%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-Pull 6088ac955775d6c0c2c240df5556d5c871171c7f parameter/ParameterHandle.cpp 111'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/276#discussion_r41889114'>File: parameter/ParameterHandle.cpp:L74-264</a></b>
- <a href='https://github.com/krocard'></a> Is this `true` the only difference between `getAsArrayOf` and `getAs` ? If so, consider merging the two.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/276#issuecomment-146866607'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard Please review
- <a href='https://github.com/dawagner'></a> @tcahuzax please review too.
- <a href='https://github.com/tcahuzax'></a> :+1:
- <a href='https://github.com/dawagner'></a> @krocard Please review.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/276?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/276?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/276?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/276'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/276'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
@krocard Please review

**Comment by dawagner** [comment]
@tcahuzax please review too.

File: parameter/ParameterHandle.cpp [hunk 1]
```diff
@@ -74,407 +74,190 @@ string CParameterHandle::getKind() const
...
+
     // Ensure we're safe against blackboard foreign access
     lock_guard<mutex> autoLock(_pParameterMgr->getBlackboardMutex());
 
-    // Define access context
-    CParameterAccessContext parameterAccessContext(strError, _pParameterMgr->getParameterBlackboard());
+    CParameterAccessContext parameterAccessContext(error, _pParameterMgr->getParameterBlackboard());
 
-    return _pBaseParameter->accessAsBooleanArray(abValues, false, parameterAccessContext);
+    return _pBaseParameter->access(values, true, parameterAccessContext);
```
Comment (krocard) [inline parameter/ParameterHandle.cpp]: Is this `true` the only difference between `getAsArrayOf` and `getAs` ? If so, consider merging the two.

**Comment by tcahuzax** [comment]
:+1:

**Comment by dawagner** [comment]
@krocard Please review.

**Comment by krocard** [comment]
:shipit:

**Pull request merged**